mysql-test/suite/innodb_fts/t/plugin_debug.test

```diff
@@ -0,0 +1,293 @@
+# Enable diag print to print query parse tree
+SET GLOBAL innodb_ft_enable_diag_print = 1;
+
+CREATE TABLE articles (
+        id INT UNSIGNED AUTO_INCREMENT NOT NULL PRIMARY KEY,
+        title VARCHAR(200),
+        body TEXT,
+        FULLTEXT (title, body) WITH PARSER simple_parser
+        ) ENGINE=InnoDB;
+
+INSERT INTO articles (title, body) VALUES
+  ('MySQL Tutorial','DBMS stands for MySQL DataBase ...'),
+  ('MySQL Tutorial','DBMS stands for MySQL good one DataBase ...'),
+  ('How To Use MySQL Well','After you went through a ...'),
+  ('Optimizing MySQL','In this tutorial we will show ...'),
+  ('1001 MySQL Tricks','How to use full-text search engine');
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+mysql -database' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('>mysql <database ~search' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+(mysql database) -engine' IN BOOLEAN MODE);
+
+# The will be no result because of parsing error.
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+(mysql database -engine' IN BOOLEAN MODE);
+
+# Test phrase match
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('"mysql"' IN BOOLEAN MODE);
+
+# Test stopword or word less than fts_min_token_size
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('"for mysql"' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('"mysql database"' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('"full text search"' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('"full text search engine"' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+"dbms stands" -good' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+("dbms stands") -good' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+("dbms stands" search) -good' IN BOOLEAN MODE);
+
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('+("dbms stands" "full text") -good' IN BOOLEAN MODE);
+
+# Test proximity match
+# It will be treated as phrase match
+SELECT * FROM articles WHERE
+        MATCH(title, body) AGAINST('("msyql database")@3' IN BOOLEAN MODE);
+
+DROP TABLE articles;
+
+SET NAMES utf8;
+-- echo # Test Part 7: Test Different Charset
+CREATE TABLE articles (
+        id INT UNSIGNED AUTO_INCREMENT NOT NULL PRIMARY KEY,
+        title VARCHAR(200),
+        body TEXT,
+        FULLTEXT (title, body) WITH PARSER simple_parser
+        ) ENGINE=InnoDB DEFAULT CHARACTER SET gb2312 COLLATE gb2312_chinese_ci;
+
+INSERT INTO articles (title, body) VALUES
+  ('数据库 是 数据 的 结构化 集合','它 可以 是 任何 东西'),
+  ('从 简单 的 购物清单 到 画展','或 企业网络 中 的 海量信息'),
+  ('要 想 将 数据 添加 到 数据库','或 访问、处理 计算机 数据库 中 保存 >的 数据'),
```

Comment:
There is no `>` in the MySQL 5.7 test file. There are also some white space differences, which I think must be reverted. The following difference is acceptable:
```diff
@@ -2,9 +2,7 @@
 --source include/have_debug.inc
 --source include/have_simple_parser.inc
 
-# Install fts parser plugin
---replace_regex /\.dll/.so/
-eval INSTALL PLUGIN simple_parser SONAME '$SIMPLE_PARSER';
+INSTALL PLUGIN simple_parser SONAME 'mypluglib';
 
 -- echo # Test Part 2: Create Index Test(CREATE TABLE WITH FULLTEXT INDEX)
 CREATE TABLE articles (
```